argilla/src/argilla/records/_dataset_records.py

```diff
@@ -357,26 +357,36 @@ def to_datasets(self) -> HFDataset:
 
     def _ingest_records(
         self,
-        records: Union[List[Dict[str, Any]], Dict[str, Any], List[Record], Record, HFDataset],
-        mapping: Optional[Dict[str, str]] = None,
+        records: Union[List[Dict[str, Any]], List[Record], HFDataset],
+        mapping: Optional[Dict[str, Union[str, Sequence[str]]]] = None,
         user_id: Optional[UUID] = None,
     ) -> List[RecordModel]:
+        """Ingests records from a list of dictionaries, a Hugging Face Dataset, or a list of Record objects."""
+
         if len(records) == 0:
             raise ValueError("No records provided to ingest.")
+
         if HFDatasetsIO._is_hf_dataset(dataset=records):
             records = HFDatasetsIO._record_dicts_from_datasets(dataset=records)
-        if all(map(lambda r: isinstance(r, dict), records)):
-            # Records as flat dicts of values to be matched to questions as suggestion or response
-            records = [self._infer_record_from_mapping(data=r, mapping=mapping, user_id=user_id) for r in records]  # type: ignore
-        elif all(map(lambda r: isinstance(r, Record), records)):
-            for record in records:
-                record.dataset = self.__dataset
-        else:
-            raise ValueError(
-                "Records should be a a list Record instances, "
-                "a Hugging Face Dataset, or a list of dictionaries representing the records."
-            )
-        return [record.api_model() for record in records]
+
+        ingested_records = []
+        record_mapper = IngestedRecordMapper(mapping=mapping, dataset=self.__dataset, user_id=user_id)
+        for record in records:
+            try:
+                if not isinstance(record, Record):
```

Comment:
we can avoid unexpected errors trying to map some weird content.
```suggestion
                if isinstance(record, dict):
```